lib/helper/Nightmare.js

```diff
@@ -206,14 +206,20 @@ class Nightmare extends Helper {
     });
   }
 
-  _withinBegin(locator) {
+  _withinBegin(locator , frames) {
     this.context = locator;
     locator = guessLocator(locator) || {css: locator};
-    this.browser.evaluate(function (by, locator) {
-      var el = codeceptjs.findElement(by, locator);
+    this.browser.evaluate(function (by, locator, frames) {
+      var el;
+      if(!frames){
+        el = codeceptjs.findElement(by, locator);
+      }
+      else{
+        el = frames.reduce((parent,child)=>parent.querySelector(child).contentDocument , window.document );
+      }
       if (!el) throw new Error(`Element by ${by}: ${locator} not found`);
```

Comment:
Tried to change findElement() and findElements() but current version of findElements required more changes. I also understand that it can take care of caching elements.

If thats really necessary changes can be made.